gobblin-data-management/src/main/java/gobblin/data/management/copy/hive/HiveLocationDescriptor.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.data.management.copy.hive;
+
+import java.io.IOException;
+import java.util.Properties;
+import java.util.Set;
+
+import lombok.Data;
+
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hive.ql.metadata.Partition;
+import org.apache.hadoop.hive.ql.metadata.Table;
+import org.apache.hadoop.mapred.InputFormat;
+
+import gobblin.data.management.copy.RecursivePathFinder;
+
+/**
+ * Contains data for a Hive location as well as additional data if {@link #ENABLE_HIVE_LOCATION_DESCRIPTOR_WITH_ADDITIONAL_DATA} set to true.
+ */
+
+@Data
+public class HiveLocationDescriptor {
+  public static final String ENABLE_HIVE_LOCATION_DESCRIPTOR_WITH_ADDITIONAL_DATA = "enable.hive.location.descriptor.with.additional.data";
+
+  protected final Path location;
+  protected final InputFormat<?, ?> inputFormat;
+  protected final FileSystem fileSystem;
+  protected final Properties properties;
+
+  public Set<Path> getPaths() throws IOException {
+    Set<Path> result = HiveUtils.getPaths(this.inputFormat, this.location);
+    
+    boolean useHiveLocationDescriptorWithAdditionalData = 
+        Boolean.valueOf(this.properties.getProperty(ENABLE_HIVE_LOCATION_DESCRIPTOR_WITH_ADDITIONAL_DATA, "true"));
+    
+    if(useHiveLocationDescriptorWithAdditionalData){
+      RecursivePathFinder finder = new RecursivePathFinder(fileSystem, location, properties);
```

Comment:
Note that location need not be a single path. It could be any glob (e.g. `/path/*`, `/path/*/foo`, or even `/path/a,/path/b`). Maybe throw an error if the path is a glob? (You can use `PathUtils.isGlob`).